sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -61,112 +177,257 @@ Table 1. Values of the `SYCL_EXT_ONEAPI_GRAPH` macro.
+When using the explicit graph building API to construct a graph, nodes and
+edges are captured as follows.
 
-[source,c++]
-----
-namespace sycl::ext::oneapi::experimental {
+Table 3. Explicit Graph Definition.
+[%header,cols="1,3"]
+|===
+| Concept | Description
 
-  class node{
-  };
-}
+| Node
+| In the explicit graph building API nodes are created by the user invoking
+methods on a modifiable graph. Each node represent either a command-group
+function, empty operation, or device memory allocation/free.
+
+| Edge
+| In the explicit graph building API edges are defined by the user. This is
+either through buffer accessors, the `make_edge()` free function, or by passing
+dependent nodes on creation of a new node.
+|===
+
+==== Queue Recording API
+
+When using the record & replay API to construct a graph by recording a queue,
+nodes and edges are captured as follows.
+
+Table 4. Recorded Graph Definition.
+[%header,cols="1,3"]
+|===
+| Concept | Description
+
+| Node
+| Nodes in a queue recorded graph represent each of the command group
+submissions of the program. Each submission encompasses either one or both of
+a.) some data movement, b.) a single asynchronous kernel launch. Nodes cannot
+define forward edges, only backwards (i.e. kernels can only create dependencies
+on things that have already happened). This means that transparently a node can
+depend on a previously recorded graph (sub-graph), which works by creating edges
+to the individual nodes in the old graph. Explicit memory operations without
+kernels, such as a memory copy, are still classed as nodes under this
+definition, as the {explicit-memory-ops}[SYCL 2020 specification states] that
+these can be seen as specialized kernels executing on the device.
+
+| Edge
+| An edge in a queue recorded graph represents a data dependency between two
+nodes. These dependencies are expressed by the user code through buffer
+accessors. There is also the partial ability to track USM data dependencies
+provided the pointers used in the graph nodes are the same. With the limitation
+that a node taking an offset USM pointer input will not be identified as having
+an edge to another node taking a pointer input to the base address of the same
+USM allocation.
+|===
+
+=== API Modifications
+
+[source, c++]
+----
+namespace sycl {
+namespace ext::oneapi::experimental {
+
+// State of a queue, returned by info::queue::state
+enum class queue_state {
+  executing,
+  recording
+};
+
+class node {};
+
+void make_edge(node sender, node receiver);
+
+// State of a graph
+enum class graph_state {
+  modifiable,
+  executable
+};
+
+// New object representing graph
+template<graph_state State = graph_state::modifiable>
+class command_graph {};
+
+template<>
+class command_graph<graph_state::modifiable> {
+public:
+  command_graph(const property_list &propList = {});
+  command_graph<graph_state::executable> finalize(const context &syclContext) const;
+
```

Comment:
```suggestion
  void make_edge(node sender, node receiver);
```

Reply:
We recently changed `make_edge` from free standing function to member function of a modifiable graph. To avoid cross graph synchronization.

Reply:
Ahh see that in the PoC branch, updated